lib/msf/ui/console/driver.rb

```diff
@@ -572,6 +564,23 @@ def handle_loglevel(val)
     set_log_level(Msf::LogSource, val)
   end
 
+  #
+  # This method handles setting a desired payload
+  #
+  # TODO: Move this out of the console driver!
+  #
+  def handle_payload(val)
```

Comment:
Yeah, not sure about this one yet, since I simply moved it out of `on_variable_set` like the others. I don't like how the console driver is doing all these things, but that's another refactor for the future.

Reply:
It should also be noted that `is_payload_compatible?` is not the same as checking `compatible_payloads`, so things like platform and arch aren't matched on.

So, it has been and still is possible to set a Linux payload on `ms17_010_eternalblue`, for instance. Not a fan of this behavior, so maybe we can leverage `compatible_payloads` here after #11768.

I'd rather do it in a separate PR.